### DISCUSSION TITLE
Allow `MEILI_HTTP_ADDR` to use a subpath

### DISCUSSION THREAD

**Discussion by mskyttner** [meilisearch/product#295] [Feedback & Feature Proposal]
Looking at https://docs.meilisearch.com/reference/features/configuration.html#http-address-port-binding and setting the MEILI_HTTP_ADDR to include a subpath ie instead of the default HTTP address and port "127.0.0.1:7700" I instead use "127.0.0.1:7700/mysubpath", the server refuses to start with an error "Error: invalid port value". 

**To Reproduce**
1. Use a docker-compose.yml file with the following service in it:

```yml
  meilisearch:
    image: getmeili/meilisearch
    env_file: ./env/meilisearch
    command: ./meilisearch --no-analytics=true
    volumes:
      - ./data.ms:/data.ms
```
2. Put an env/meilisearch file with environment variables like this:

```bash
MEILI_HTTP_ADDR=127.0.0.1:7700/mysubpath
MEILI_MASTER_KEY=myMasterkey
```
3. Start with "docker-compose up -d" and check logs with "docker-compose logs -f  meilisearch"

4. See error in log

**Expected behavior**

Initially startup looked promising in the log with "Server listening on:	"http://127.0.0.1:7700/mysubpath". But then "Error:invalid port value" appeared and the server did not start.

I expected the MEILI_HTTP_ADDR setting to allow for a subpath which would enable reverse proxying into a subpath application using nginx or caddy or other reverse proxy services. 

** Rationale **

It would be very convenient to be able to configure the base_url for the service (API and web GUI) so it can easier run alongside with other services on the same subdomain, and easily be reverse proxied into a subpath using simple nginx or caddy directives. 

This describes the issue "why can't I reverse proxy my app into a subfolder?"-issue well: https://caddy.community/t/the-subfolder-problem-or-why-cant-i-reverse-proxy-my-app-into-a-subfolder/8575

**MeiliSearch version:** [e.g. v0.23.1]

**Comment by curquiza** [comment]
Hello @mskyttner 

Thanks for this feature request. I moved it to the dedicated repository for the feature requests. The product team will consider your suggestion. Thanks for your feedback :)

**Reply by mskyttner** [reply]
Relative URLs in mini-dashboard would be needed for this, I guess?

https://github.com/meilisearch/mini-dashboard/issues/236.

**Reply by tunchunairarko** [reply]
Any update on this, please?

**Reply by macraig** [reply]
Hi @tunchunairarko , we have no updates on this for now.
